dist/js/select2.full.js

```diff
@@ -1802,7 +1802,11 @@ S2.define('select2/selection/allowClear',[
       }
     }
 
-    this.$element.val(this.placeholder.id).trigger('change');
+    if (typeof this.placeholder !== 'undefined') {
+        this.$element.val(this.placeholder.id).trigger('change');
+    } else {
+        this.$element.val('').trigger('change');
+    }
```

Comment:
Change in dist/ better left off: https://github.com/select2/select2/blob/master/CONTRIBUTING.md#submitting-a-pull-request